[10.18 release] Edge-CIE-5182 10.17 to 10.18 Migration of Edge Appliance (#1931)
SoftwareAG/c8y-docs: content/edge/update-bundle/migration-1017-to-1018.md

---
weight: 15
title: Migrating Edge 10.17 to 10.18
layout: redirect
---

The following section outlines the procedure for migrating from Edge 10.17 to 10.18. Since Edge 10.18 is based on the Rocky Linux operating system, the upgrade process entails a side-by-side installation accompanied by data migration.

{{< c8y-admon-important >}} 
If you are using a version earlier than Edge 10.17 and plan to migrate to Edge 10.18, you must first update your Edge to version 10.17.
If you are using a version prior to Edge 10.13, you must first update your Edge to version 10.13 before upgrading to 10.17.
{{< /c8y-admon-important >}}

For information about upgrading from an earlier version to Edge 10.17, see:

- [Updating Edge using the GUI](https://{{< domain-c8y >}}/guides/10.17.0/edge/update/#updating-edge-gui)
- [Updating Edge using the REST APIs](https://{{< domain-c8y >}}/guides/10.17.0/edge/update/#updating-edge-rest)

in the *10.17.0 {{< product-c8y-iot >}} Edge guide*.

## Backing Up Data and Configuration on Edge 10.17 {#backing-up-data-on-edge-1017}
In your Edge 10.17 appliance, back up the MongoDB data, Edge, Cumulocity Agent, and OPC UA configurations.

> {{< c8y-admon-important >}} Before attempting the backup, ensure that there is sufficient disk space to save the backup in your Edge 10.17 appliance.  

Reviewer: ```suggestion
{{< c8y-admon-important >}}
Before attempting the backup, ensure that there is sufficient disk space to save the backup in your Edge 10.17 appliance.  
```

{{< /c8y-admon-important >}}

Perform the following steps as a `root` user on your Edge 10.17 appliance.

1. Run the following commands to unmonitor all services and stop the `opcua-mgmt-service`, `opcua-device-gateway`, `smartrule`, `apama`, `cumulocity-core-karaf` and `mongod` services:

   ```shell
   monit unmonitor all
   systemctl stop installation-service opcua-mgmt-service opcua-device-gateway smartrule apama cumulocity-core-karaf mongod
   ```

> 2. Back up the MongoDB data, data lake contents from the DataHub if present, Edge, Cumulocity Agent, and OPC UA configurations using the following command to create the `/opt/edge-1017-backup.tar` file:

Reviewer: "Cumulocity Agent" --> Cumulocity as the product (= Cumulocity IoT) or Cumulocity as the company name?

Author: **Cumulocity Agent** is a technical/internal component name which would not change when the product/doc is rebranded. We still have to refer to this component in the doc as "Cumulocity Agent".
@MWindel , I am marking this topic resolved based on our discussion today.

   ```shell
   tar -zcf /opt/edge-1017-backup.tar /opt/softwareag /opt/mongodb /var/lib/cumulocity-agent /usr/edge /opt/opcua 
   ```

3. After creating the `/opt/edge-1017-backup.tar` file, copy it to a network drive or storage location that is accessible from the Edge 10.18 appliance you will create in the next step. Once the backup file is safely stored, shut down the Edge 10.17 appliance to prevent any further changes to the system during the migration process. This step is optional and if not performed, you need to copy it directly into the Edge 10.18 appliance once it is created. 

## Installing Edge 10.18 {#installing-edge-1018}
Install the Edge appliance 10.18 following the instructions below. 

> {{< c8y-admon-important >}} If you have both the Edge 10.17 and Edge 10.18 appliances on the same host machine, ensure that the IP addresses of both appliances are different.{{< /c8y-admon-important >}}

Reviewer: ```suggestion
{{< c8y-admon-important >}}
If you have both the Edge 10.17 and Edge 10.18 appliances on the same host machine, ensure that the IP addresses of both appliances are different.
{{< /c8y-admon-important >}}
```

4. If you have installed {{< product-c8y-iot >}} DataHub in the Edge 10.17 appliance, follow the instructions in [Setting up {{< product-c8y-iot >}} DataHub Edge](/datahub/running-datahub-on-the-edge/#setting-up-datahub-edge) to install the same in Edge 10.18. Before you proceed with installing {{< product-c8y-iot >}} DataHub, change the ownership of the data lake folders using the following command:
   ```shell
   chown -R systemd-coredump:systemd-coredump /opt/mongodb/cdh-*
   ```

5. For information about security configuration, refer to [Configuring security](/edge/configuration/#configuring-security).

For detailed guidance on configuring the Edge 10.18 appliance, consult [Configuring Edge](/edge/configuration/).

## Sample script to automate the data migration {#sample-script-to-automate-data-migration}
{{< company-sag >}} provides the `edge-1017-to-1018-data-migration.sh` script as a reference for automating the data migration process. You can customize this script according to your requirements. Click [here](/files/edge/edge-1017-to-1018-data-migration.sh) to download the script.

> {{< c8y-admon-important >}}{{< company-sag >}} does not officially support this migration script. It is provided for reference purposes only.{{< /c8y-admon-important >}}

Reviewer: ```suggestion
{{< c8y-admon-important >}}
{{< company-sag >}} does not officially support this migration script. It is provided for reference purposes only.
{{< /c8y-admon-important >}}
```

---
weight: 15
title: Migrating Edge 10.17 to 10.18
layout: redirect
---

The following section outlines the procedure for migrating from Edge 10.17 to 10.18. Since Edge 10.18 is based on the Rocky Linux operating system, the upgrade process entails a side-by-side installation accompanied by data migration.

{{< c8y-admon-important >}} 
If you are using a version earlier than Edge 10.17 and plan to migrate to Edge 10.18, you must first update your Edge to version 10.17.
If you are using a version prior to Edge 10.13, you must first update your Edge to version 10.13 before upgrading to 10.17.
{{< /c8y-admon-important >}}

For information about upgrading from an earlier version to Edge 10.17, see:

- [Updating Edge using the GUI](https://{{< domain-c8y >}}/guides/10.17.0/edge/update/#updating-edge-gui)
- [Updating Edge using the REST APIs](https://{{< domain-c8y >}}/guides/10.17.0/edge/update/#updating-edge-rest)

in the *10.17.0 {{< product-c8y-iot >}} Edge guide*.

## Backing Up Data and Configuration on Edge 10.17 {#backing-up-data-on-edge-1017}

> In your Edge 10.17 appliance, back up the MongoDB data, Edge, Cumulocity Agent, and OPC UA configurations.

Reviewer: ```suggestion
In your Edge 10.17 appliance, back up the MongoDB data, Edge, {{< company-c8y >}} Agent, and OPC UA configurations.
```
